bin/si-conduit/src/analytics.ts

```diff
@@ -0,0 +1,83 @@
+import { PostHog } from "posthog-node";
+
+const POSTHOG_EVENT_PREFIX = "conduit";
+const POSTHOG_SHUTDOWN_MAX_MS = 2000;
+
+export class Analytics {
+  private posthog: PostHog;
+  private readonly sessionId: string;
+
+  private readonly userId: string | undefined;
+  private readonly workspaceId: string | undefined;
+
+  constructor(authData?: { userId: string; workspaceId: string }) {
+    this.sessionId = crypto.randomUUID();
+    this.posthog = this.initializePostHog();
+
+    this.userId = authData?.userId;
+    this.workspaceId = authData?.workspaceId;
+
+    if (!this.userId) return;
+    this.posthog.identify({
+      distinctId: this.userId,
+      properties: {
+        userId: this.userId,
+      },
+    });
+  }
+
+  private initializePostHog(): PostHog {
+    const apiKey = Deno.env.get("POSTHOG_API_KEY") ||
+      "phc_KpehlXOqtU44B2MeW6WjqR09NxRJCYEiUReA58QcAYK"; // Prod Posthog
```

Comment:
We should post that this is a publicly used API Key or we will get reports from users